bot/converters.py

```diff
@@ -567,6 +643,26 @@ async def convert(self, ctx: Context, arg: str) -> t.Optional[dict]:
             return await ctx.bot.api_client.get(f"bot/infractions/{arg}")
 
 
+if t.TYPE_CHECKING:
+    ValidDiscordServerInvite = dict  # noqa: F811
+    ValidFilterListType = str  # noqa: F811
+    Extension = str  # noqa: F811
+    PackageName = str  # noqa: F811
+    ValidURL = str  # noqa: F811
+    Inventory = t.Tuple[str, _inventory_parser.InventoryDict]  # noqa: F811
+    Snowflake = int  # noqa: F811
+    TagNameConverter = str  # noqa: F811
+    TagContentConverter = str  # noqa: F811
+    SourceConverter = SourceType  # noqa: F811
+    DurationDelta = relativedelta  # noqa: F811
+    Duration = datetime  # noqa: F811
+    OffTopicName = str  # noqa: F811
+    ISODateTime = datetime  # noqa: F811
+    HushDurationConverter = int  # noqa: F811
+    UserMentionOrID = discord.User  # noqa: F811
+    FetchedUser = t.Union[discord.User, discord.Object]  # noqa: F811
+    Infraction = t.Optional[dict]  # noqa: F811
```

Comment:
For what it's worth, `bot.exts.moderation.infractions._utils.Infraction` exists. I don't think being slightly more specific with the dict types really adds much when it comes to JSON though.

Reply:
I think it could be worth it with a TypedDict in the infractions package